forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-rcs/forgerock-openbanking-uk-aspsp-rs-rcs-server/src/main/java/com/forgerock/openbanking/aspsp/rs/rcs/api/rcs/autoaccept/AutodecisionsApiController.java

```diff
@@ -90,8 +92,9 @@
 
             log.debug("Read payment ID from the claims");
             //Read the claims
-            Claims claims = Claims.parseClaims(signedJWT.getJWTClaimsSet().getJSONObjectClaim(OIDCConstants
-                    .OIDCClaim.CLAIMS));
+            JWTClaimsSet claimSet = signedJWT.getJWTClaimsSet();
+            Map<String, Object> jsonClaimSet = claimSet.getJSONObjectClaim(OIDCConstants.OIDCClaim.CLAIMS);
+            Claims claims = Claims.parseClaims(new JSONObject(jsonClaimSet));
```

Comment:
I think it's worth putting this in a helper class so that it's not repeated everywhere.

Reply:
Yes. Good idea.